### PR TITLE
replace exit with Exitcode

### PR DESCRIPTION
**Pull request by joohongpark**
issue #3726

### CODE REVIEW

File: examples/generator.rs [hunk 1]
```diff
@@ -40,10 +41,10 @@ gen()
...
 }
 
-fn main() {
+fn main() -> ExitCode {
     let interp = vm::Interpreter::with_init(Default::default(), |vm| {
         vm.add_native_modules(rustpython_stdlib::get_module_inits());
     });
     let result = py_main(&interp);
-    std::process::exit(interp.run(|_vm| result));
+    ExitCode::from(interp.run(|_vm| result).to_be_bytes()[0])
```
Comment (youknowone) [inline examples/generator.rs:49]: I don't think we need `to_be_bytes()`. the exitcode was already int.

**Comment by joohongpark** [comment]
The work that is difficult to reflect immediately has a TODO comment.

File: parser/build.rs [hunk 2]
```diff
@@ -2,15 +2,19 @@ use std::fmt::Write as _;
...
+    let result = check_lalrpop("src/python.lalrpop", "src/python.rs");
+    if let Some(result) = result {
+        return result;
+    }
     gen_phf();
+    ExitCode::SUCCESS
 }
 
-fn check_lalrpop(source: &str, generated: &str) {
+fn check_lalrpop(source: &str, generated: &str) -> Option<ExitCode> {
```
Comment (youknowone) [inline parser/build.rs:17]: I think this function is expected to return Result instead of Option<ExitCode>

File: stdlib/src/posixsubprocess.rs [hunk 3]
```diff
@@ -92,6 +92,7 @@ fn exec(args: &ForkExecArgs, procargs: ProcArgs) -> ! {
             let _ = write!(cur, "OSError:{}:", e as i32);
             let pos = cur.position();
             let _ = unistd::write(args.errpipe_write, &buf[..pos as usize]);
+            // TODO: Need to change to ExitCode or Termination
```
Comment (youknowone) [inline stdlib/src/posixsubprocess.rs:95]: I don't think this is expected to be ExitCode because it doesn't raise SystemExit

**Review by youknowone — CHANGES_REQUESTED** [review]
I left a few comments, but other parts looks great!

File: parser/build.rs (hunk 2 above)
Comment (joohongpark) [reply parser/build.rs:17]: There is a control statement that returns `None` in the same function, so isn't it possible to reach that code?

Comment (youknowone) [reply parser/build.rs:17]: Now, error case is encoded as `Some(ExitCode)` and successful case is encoded as `None`. This is typically encoded as Result type in Rust code like `Ok(())` and `Err(_)`. Give please give a proper error to `Err` case.

Comment (youknowone) [reply parser/build.rs:17]: Edit: because error type is not what this PR covers, `Result<(), ExitCode>` will be good enough.

**Review by youknowone — APPROVED** [review]
looks good, thank you!

**Pull request merged**